### ISSUE TITLE
incorrect line within code block

### ISSUE THREAD

**Issue by seoh** [hexojs/hexo#1002]
if there is code like this
```swift
println(1);

println(2);

println(3);
```

then it will be presented

``` text
1
2 println(1);
3 println(2);
4 println(3);
5
```

It is caused by blank lines of `<div class="line"></div>`
I think two solutions.

#####  1. simply, give them height of one line.

in lib/util/highlight.js:94(in hexo v2.8.3, but it will be fixed in 3.0 as https://github.com/hexojs/util/blob/master/lib/highlight.js#L32) 
``` js
content += '<div class="line">' + item + '</div>';
```

will be changed to

```
content += '<div class="line">' + (item || '&nbsp;') + '</div>';
```

##### 2. pseudo class of one line in theme
``` css
.line::before {
content: '';
display: inline-block;
}
```

I'm not sure which is better.

**Comment by iissnan** [comment]
I prefer solution 1.
For option 2, using CSS in not safe enough, because it can be overrode by themes.

By the way, the fix code in solution 1 misses a single quote after `&nbsp;`.

**Comment by seoh** [comment]
oops. fixed typo. I also prefer 1, and that is what applied in my local env.

**Comment by iissnan** [comment]
Hi @seoh, maybe you can issue a pull request for this issue to benefit anyone else encounter this issue :)

**Comment by seoh** [comment]
It will be fixed since 3.0.(changed `div` to `span` for making line.). It just affected until v2.8.3(currently stable version). I just give a hint to the impatient(like me).

**Comment by leesei** [comment]
Related: #774, #792.

**Comment by leesei** [comment]
Fixed in 3.0.
